openpype/modules/sync_server/launch_hooks/pre_copy_last_published_workfile.py

```diff
@@ -184,3 +185,53 @@ def execute(self):
         self.data["last_workfile_path"] = local_workfile_path
         # Keep source filepath for further path conformation
         self.data["source_filepath"] = last_published_workfile_path
+
+        # Get and make resources directory
+        resources_dir = os.path.join(
+            os.path.dirname(local_workfile_path), 'resources'
+        )
+        if not os.path.exists(resources_dir):
+            os.mkdir(resources_dir)
+
+        # Copy resources to the local resources directory
+        for file in workfile_representation['files']:
+            resource_main_path = file["path"].replace(
+                "{root[main]}", str(anatomy.roots["main"])
+            )
+
+            # Only copy if the resource file exists, and it's not the workfile
+            if (
+                os.path.exists(resource_main_path)
+                and resource_main_path != last_published_workfile_path
+            ):
```

Comment:
Instead of putting everything under a single condition, I'd create a `if not ... : continue`